EventSubscriber/FrontendControllerSubscriber.php

```diff
@@ -82,6 +83,12 @@ public function onKernelRequest(GetResponseEvent $event): void
                 $frontendController->settingLanguage();
                 $frontendController->settingLocale();
             }
+
+            if (!$frontendController->tmpl->setup && ($site = $event->getRequest()->attributes->get('site')) instanceof SiteInterface) {
+                $frontendController->id = $site->getRootPageId();
+                $frontendController->determineId();
+                $frontendController->getConfigArray();
+            }
```

Comment:
Is the order correct, to determine it after language or should it be moved up?

Reply:
You are right, initialization oder is different in typo3 middlewares. Fixed it.